src/ResultSet.php

```diff
@@ -0,0 +1,74 @@
+<?php
+namespace CCR\Ralph;
+
+use IteratorAggregate, JsonSerializable;
+use function json_decode;
+
+/**
+ * A result set for working with results returned from a SPARQL endpoint.
+ */
+
+class ResultSet implements IteratorAggregate, JsonSerializable
+{
+    /** @var array Results returned from a query. */
+    private $results;
+
+    public function __construct(string $results)
+    {
+        $this->results = json_decode($results, true);
```

Comment:
Do you want to check that `json_decode` didn't return `null` or is that handled elsewhere?